spacy_llm/tasks/entity_linking.py

```diff
@@ -0,0 +1,310 @@
+from .templates import read_template
+from .util import SerializableTask
+
+_DEFAULT_EL_TEMPLATE_V1 = read_template("entity_linking")
+
+
+class EntityLinkingExample(BaseModel):
+    text: str
+    mentions_str: str
+    mentions: List[str]
+    entity_descriptions: List[List[str]]
+    entity_ids: List[List[str]]
+    solutions: List[str]
+    reasons: Optional[List[str]]
+
+
+@registry.llm_misc("spacy.CandidateSelectorPipeline.v1")
+class SpaCyPipelineCandidateSelector:
+    """Callable generated by loading and wrapping a spaCy pipeline with a NEL component and a filled knowledge base."""
+
+    def __init__(
+        self,
+        nlp_path: Union[Path, str],
+        desc_path: Union[Path, str],
+        top_n: int = 5,
+    ):
+        """
+        Loads spaCy pipeline, knowledge base, entity descriptions.
+        top_n (int): Top n candidates to include in prompt.
+        nlp_path (Union[Path, str]): Path to stored spaCy pipeline.
+        desc_path (Union[Path, str]): Path to .csv file with descriptions for entities. Has to have two columns
+          with the first one being the entity ID, the second one being the description. The entity ID has to match with
+          the entity ID in the stored knowledge base.
+        """
+        self._nlp = spacy.load(nlp_path)
+        if "entity_linker" not in self._nlp.component_names:
+            raise ValueError(
+                f"'entity_linker' component has to be available in specified pipeline at {nlp_path}, but "
+                f"isn't."
+            )
+        self._entity_linker: EntityLinker = self._nlp.get_pipe("entity_linker")
+        self._kb = self._entity_linker.kb
+        with open(desc_path) as csvfile:
+            self._descs: Dict[str, str] = {}
+            for row in csv.reader(csvfile, quoting=csv.QUOTE_ALL, delimiter=";"):
+                if len(row) != 2:
+                    continue
+                self._descs[row[0]] = row[1]
+        self._top_n = top_n
+
+    def __call__(self, mentions: Iterable[Span]) -> Iterable[Dict[str, str]]:
+        """Retrieves top n candidates using spaCy's entity linker's .get_candidates_batch().
+        mentions (Iterable[Span]): Mentions to look up entity candidates for.
+        RETURNS (Iterable[Dict[str, str]]): Dicts of entity ID -> description for all candidates, per mention.
+        """
+        all_cands = self._kb.get_candidates_batch(mentions)
+        for cands in all_cands:
+            assert isinstance(cands, list)
+            cands.sort(key=lambda x: x.prior_prob, reverse=True)
+
+        return [
+            {cand.entity_: self._descs[cand.entity_] for cand in cands[: self._top_n]}
+            for cands in all_cands
+        ]
+
+    def get_entity_description(self, entity_id: str) -> str:
+        if entity_id not in self._descs:
+            raise ValueError(
+                f"Entity with ID {entity_id} is not in provided descriptions file."
+            )
```

Comment:
I'm not convinced yet that this should be an error. I imagine there could be a KB that has descriptions for 90% of the entities and no description for some, in which case the LLM won't have any information to go by, but it feels wrong to crash in these cases, especially if the right entity does have a description?

Reply:
Fair enough. In this case we'd present an entity without description to the LLM, which is likely also not wanted by the user. Changed to warning in https://github.com/explosion/spacy-llm/pull/241/commits/c3d7a91c8fef1e3f59d207e1c58de28880a2437b.